packages/eslint-plugin/src/rules/no-unnecessary-condition.ts

```diff
@@ -26,10 +26,30 @@ import {
 
 // Truthiness utilities
 // #region
+const valueIsPseudoBigInt = (
+  value: number | string | ts.PseudoBigInt,
+): value is ts.PseudoBigInt => {
+  return typeof value === 'object' && 'base10Value' in value;
```

Comment:
Granted the input is `string | number | ts.PseudoBigInt`, checking anything but the `typeof` is unnecessary to discriminate between them
```suggestion
  return typeof value === 'object'
```